src/algorithms/pathfinders/path.ts

```diff
@@ -0,0 +1,33 @@
+import EdgeCost from '../models/edgecost';
+import Vertex from '../models/vertex';
+/**
+ * Path class, stores pathData and costs in a path
+ */
+export default class Path {
+    pathData: Array<{vertex: Vertex, edgecost: EdgeCost}>;
```

Comment:
I find these inline object definitions confusing and I think that they could always be replaced for a class or interface (like you did with IEntry). This vertex&edgecost pair is used on multiple places (at least here and in vertex.ts) and in cases like this one (type definition) it would make the code shorter and more readable.